<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>690</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>780</width>
    <height>680</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PCA9685 Servo Driver Calibration</string>
  </property>
  <property name="windowIcon">
   <iconset resource="images.qrc">
    <normaloff>:/logo/Logo_1x_White.png</normaloff>:/logo/Logo_1x_White.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QFrame" name="frame">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>181</width>
              <height>261</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <widget class="QWidget" name="verticalLayoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>160</width>
               <height>241</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>I2C Bus:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cbBusSelect"/>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>PCA9685 Address:</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>0x</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLineEdit" name="leAddressHex">
                  <property name="inputMask">
                   <string>&gt;HH</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>0b</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="leAddressBin">
                  <property name="inputMask">
                   <string>BBBBBBBB</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="btConnect">
                <property name="text">
                 <string>Connect</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btDisconnect">
                <property name="text">
                 <string>Disconnect</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="images.qrc">:/logo/Logo_1x.png</pixmap>
            </property>
            <property name="scaledContents">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QFrame" name="frame_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>410</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <widget class="QLabel" name="lImage">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>110</y>
             <width>401</width>
             <height>241</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="pixmap">
            <pixmap resource="images.qrc">:/car/images/Car.png</pixmap>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QLabel" name="lDir1">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>80</y>
             <width>111</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="pixmap">
            <pixmap resource="images.qrc">:/car/images/Arrow_Right.png</pixmap>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QLabel" name="lDir2">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>350</y>
             <width>111</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="pixmap">
            <pixmap resource="images.qrc">:/car/images/Arrow_Right.png</pixmap>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QSlider" name="slidSpeed">
           <property name="geometry">
            <rect>
             <x>140</x>
             <y>39</y>
             <width>160</width>
             <height>29</height>
            </rect>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
          <widget class="QSlider" name="slidSteer">
           <property name="geometry">
            <rect>
             <x>440</x>
             <y>150</y>
             <width>29</width>
             <height>160</height>
            </rect>
           </property>
          </widget>
          <widget class="QSpinBox" name="sBSpeedBot">
           <property name="geometry">
            <rect>
             <x>57</x>
             <y>40</y>
             <width>71</width>
             <height>27</height>
            </rect>
           </property>
          </widget>
          <widget class="QSpinBox" name="sBSpeedTop">
           <property name="geometry">
            <rect>
             <x>310</x>
             <y>40</y>
             <width>71</width>
             <height>27</height>
            </rect>
           </property>
          </widget>
          <widget class="QSpinBox" name="sBSteerTop">
           <property name="geometry">
            <rect>
             <x>430</x>
             <y>106</y>
             <width>71</width>
             <height>27</height>
            </rect>
           </property>
          </widget>
          <widget class="QSpinBox" name="sBSteerBot">
           <property name="geometry">
            <rect>
             <x>430</x>
             <y>325</y>
             <width>71</width>
             <height>27</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="lSpeedVal">
           <property name="geometry">
            <rect>
             <x>190</x>
             <y>64</y>
             <width>67</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>500</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
          <widget class="QLabel" name="lSteerVal">
           <property name="geometry">
            <rect>
             <x>389</x>
             <y>220</y>
             <width>51</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>500</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
          <widget class="QCheckBox" name="cbInvSpeed">
           <property name="geometry">
            <rect>
             <x>240</x>
             <y>13</y>
             <width>51</width>
             <height>22</height>
            </rect>
           </property>
           <property name="text">
            <string>Inv</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="cbInvSteer">
           <property name="geometry">
            <rect>
             <x>475</x>
             <y>230</y>
             <width>51</width>
             <height>22</height>
            </rect>
           </property>
           <property name="text">
            <string>Inv</string>
           </property>
          </widget>
          <widget class="QSpinBox" name="sbChannelSpeed">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>10</y>
             <width>48</width>
             <height>27</height>
            </rect>
           </property>
           <property name="maximum">
            <number>15</number>
           </property>
          </widget>
          <widget class="QLabel" name="label_5">
           <property name="geometry">
            <rect>
             <x>150</x>
             <y>15</y>
             <width>31</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>CH:</string>
           </property>
          </widget>
          <widget class="QSpinBox" name="sbChannelSteer">
           <property name="geometry">
            <rect>
             <x>505</x>
             <y>195</y>
             <width>48</width>
             <height>27</height>
            </rect>
           </property>
           <property name="maximum">
            <number>15</number>
           </property>
          </widget>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>475</x>
             <y>200</y>
             <width>31</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>CH:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_8">
           <property name="geometry">
            <rect>
             <x>421</x>
             <y>13</y>
             <width>41</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Freq:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="sBFreq">
           <property name="geometry">
            <rect>
             <x>468</x>
             <y>10</y>
             <width>91</width>
             <height>27</height>
            </rect>
           </property>
           <property name="decimals">
            <number>3</number>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextBrowser" name="tbLog">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>780</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>slidSteer</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>lSteerVal</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>649</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slidSpeed</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>lSpeedVal</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>127</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
